### PR TITLE
Vstest task changes for testselector latest changes

### PR DESCRIPTION
**Pull request by prawalagarwal**
TestSelector tool has been modified to support file level data collection in addition to method level. This PR contains the corresponding task level changes.

### CODE REVIEW

File: Tasks/VsTest/vstest.ts [hunk 1]
```diff
@@ -1015,3 +1035,11 @@ function isTiaAllowed(): boolean {
     }
     return false;
 }
+
+function isFileLevelAllowed()
```
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: >isFileLevelAllowed [](start = 9, length = 18)

the function name suggest that it shuld return bool. can we rename it?

File: Tasks/VsTest/vstest.ts [hunk 2]
```diff
@@ -332,7 +349,7 @@ function runVStest(testResultsDirectory: string, settingsFile: string, vsVersion
     }
     else {
         tl.debug("Non TIA mode of test execution");
-        executeVstest(testResultsDirectory, settingsFile, vsVersion, getVstestArguments(settingsFile))
+        executeVstest(testResultsDirectory, settingsFile, vsVersion, getVstestArguments(settingsFile, false))
```
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: >)) [](start = 107, length = 2)

do we have named params in ts?

**Comment by acesiddhu** [comment]
> function createRunSettingsForTestImpact(vsVersion: number, settingsFile: string, exitErrorMessage: string): Q.Promise<string> {

question. do we try to change the setting file if test impact node is already there?

---
Refers to: Tasks/VsTest/vstest.ts:766 in ccc49e8. [](commit_id = ccc49e86b03daaca0830042e952b63d2e8fd1208, deletion_comment = False)

File: Tasks/VsTest/vstest.ts (hunk 2 above)
Comment (prawalagarwal) [inline Tasks/VsTest/vstest.ts]: Nope. Not as of now


---
In reply to: [74879634](https://github.com/Microsoft/vsts-tasks/pull/2264#discussion_r74879634) [](ancestors = 74879634)

**Comment by prawalagarwal** [comment]
> function createRunSettingsForTestImpact(vsVersion: number, settingsFile: string, exitErrorMessage: string): Q.Promise<string> {

We were not checking for inner nodes like Impact level and root path. Added them now.

---
In reply to: [240009732](https://github.com/Microsoft/vsts-tasks/pull/2264#issuecomment-240009732) [](ancestors = 240009732)

---
Refers to: Tasks/VsTest/vstest.ts:766 in ccc49e8. [](commit_id = ccc49e86b03daaca0830042e952b63d2e8fd1208, deletion_comment = False)

**Comment by acesiddhu** [comment]
:shipit:

**Pull request merged**